src/main/java/org/mockito/internal/matchers/CapturingMatcher.java

```diff
@@ -15,13 +15,13 @@
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
 import org.mockito.ArgumentMatcher;
+import org.mockito.internal.util.Primitives;
 
-@SuppressWarnings("unchecked")
 public class CapturingMatcher<T>
         implements ArgumentMatcher<T>, CapturesArguments, VarargMatcher, Serializable {
```

Comment:
As a follow-up cleanup, I think we can also remove the `CapturesArguments` interface, since this is the only implementation. Then we can match on `CapturingMatcher` in https://github.com/mockito/mockito/blob/123beb8311ed6303ba71d682cff8434fe36b7f0f/src/main/java/org/mockito/internal/invocation/InvocationMatcher.java#L143